test/pycheck/motherduck_test.py

```diff
@@ -56,6 +56,37 @@ def test_md_create_table(md_cur: Cursor, ddb):
         md_cur.sql("CREATE TABLE t4(a int) USING heap")
 
 
+def test_md_read_scaling(pg: Cursor, ddb, default_db_name, md_test_user):
+    weird_db_name = "some 19 really $  - @ weird name 😀 84"
```

Comment:
I feel like this test tests two very separate things at the same time. Which makes it hard to follow what's happening. Let's split it up in a test that tests that the escaping of the default database. And one that tests the session hint behavior.

I also think we need to add a bunch of other read scaling tests. But this seems like a nice start at least. And the PR fixes an important bug, so let's merge it quickly.